### ISSUE TITLE
[NEXT-1192] Failed to find font override - next/font/google

### ISSUE THREAD

**Issue by maxencerb** [vercel/next.js#47115]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.3.0: Mon Jan 30 20:38:37 PST 2023; root:xnu-8792.81.3~2/RELEASE_ARM64_T6000
    Binaries:
      Node: 18.14.0
      npm: 9.3.1
      Yarn: 1.22.19
      pnpm: N/A
    Relevant packages:
      next: 13.2.5-canary.3
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

Font optimization (@next/font)

### Link to the code that reproduces this issue

https://github.com/maxencerb/next-font-bug

### To Reproduce

```ts utils/fonts.ts
// utils/fonts.ts
import { Unbounded } from 'next/font/google';

export const unbounded = Unbounded({
  subsets: ['latin'],
  display: 'swap',
})
```

```tsx app/page.tsx
// app/page.tsx
import { unbounded } from "../utils/fonts"

export default function Home() {
  return (
    <h1 className={unbounded.className}>TEST TITLE</h1>
  )
}
```

## Output

Output works as expected

![image](https://user-images.githubusercontent.com/47566657/224982403-c0981374-36dc-491c-ad36-3fa306203dce.png)

## Console Output

```bash
error - Failed to find font override values for font `Unbounded`
```

## Explaination

All fallback font metrics from google are loaded from `next/font/google/get-fallback-font-override-metrics.ts` where we can see the following method and doc :

```ts
// @ts-ignore
import { calculateSizeAdjustValues } from 'next/dist/server/font-utils'
// @ts-ignore
import * as Log from 'next/dist/build/output/log'

/**
 * Get precalculated fallback font metrics for the Google Fonts family.
 *
 * TODO:
 * We might want to calculate these values with fontkit instead (like in next/font/local).
 * That way we don't have to update the precalculated values every time a new font is added to Google Fonts.
 */
export function getFallbackFontOverrideMetrics(fontFamily: string) {
  try {
    const { ascent, descent, lineGap, fallbackFont, sizeAdjust } =
      calculateSizeAdjustValues(
        require('next/dist/server/google-font-metrics.json')[fontFamily]
      )
    return {
      fallbackFont,
      ascentOverride: `${ascent}%`,
      descentOverride: `${descent}%`,
      lineGapOverride: `${lineGap}%`,
      sizeAdjust: `${sizeAdjust}%`,
    }
  } catch {
    Log.error(`Failed to find font override values for font \`${fontFamily}\``)
  }
}
```

So I think there might be a fix but I don't really know how to implement it. I tried looking at the local font optimization code.

Maybe someone can either find the solution to calculate the values with fontkit or at least update the json file under `next/src/server/google-font-metrics.json` and tell how to find those values.

### Expected Behavior

Find the font fallback for any google font

<sub>[NEXT-1192](https://linear.app/vercel/issue/NEXT-1192/failed-to-find-font-override-nextfontgoogle)</sub>

**Comment by joepetrillo** [comment]
Having the same issue with `Golos_Text`

**Comment by nattwasm** [comment]
Same issue with `Newsreader`

**Comment by amol909** [comment]
same issue with `Inter_Tight`

**Comment by efollette** [comment]
same issue with `Inter_Tight`

**Comment by marcofranssen** [comment]
Same with `Sofia Sans Semi Condensed`

**Comment by jacobhq** [comment]
May be unhelpful but I just update the import to `next/font/google` (which I know you've already done) and restarted the dev server and it seemed to work again.

**Comment by Artsiom-Mazurkevich** [comment]
Hi everybody! Adding " display: 'swap' " in options, helped me!

**Comment by BMscis** [comment]
Adding @font-face on my css fixed it for me:

```
@font-face {
  font-family: 'Bungee Spice';
  src: url('/fonts/BungeeSpice-Regular.woff2') format('woff2');
  font-weight: normal;
  font-style: normal;
  font-display: swap;
}
```

**Comment by chris-monahan** [comment]
Seem to be having the same issue with `Alkatra`

**Comment by marcelo-albuquerque** [comment]
> Having the same issue with `Golos_Text`

Me too.

**Comment by IntiSilva** [comment]
Same issue with Bruno Ace

**Comment by IntiSilva** [comment]
Another thing, I'm seeing that a lot of people including myself are having issues with Next new appdir in the different steps of developing an app. I would recommend that u don't use the app dir and use the common pages structure, I'm using v 13.4.1 that it's supposed to be stable and I'm breaking my head against the keyboad bc I can't get a font or a simple animation to work.

**Comment by mirsahib** [comment]
Same issue with Comme
using ` "next": "13.3.4",`

**Comment by gugiserman** [comment]
Same with Texturina

**Comment by hydronatron** [comment]
> Another thing, I'm seeing that a lot of people including myself are having issues with Next new appdir in the different steps of developing an app. I would recommend that u don't use the app dir and use the common pages structure, I'm using v 13.4.1 that it's supposed to be stable and I'm breaking my head against the keyboad bc I can't get a font or a simple animation to work.

I've also been bashing my ahead against my keyboard, it's a real struggle after using the old system for so long 😅.  Take this with a grain of salt as I've been very optimistic (and enthusiastic!) about this change for a while now, but:

`/app` is a massive shift in the Next paradigm and many other components in the ecosystem haven't quite adapted to the new format. In my opinion, the App directory is a lot faster and is just *so much more convenient* to use! Personally, I make a lot of apps that pull data from APIs or databases, and the appDir just makes my life easier. Change usually takes a while, esp with smth as big as this, so it might take a while for third-party components and modules to become compatible with the new system.

In the mean time, you could wrap client-designed/legacy components in client component wrappers to have them work with the new system - I think the Next docs even talk about doing this (although I may be wrong).

Also, still having this issue with `Golos_Text` as of today on 13.4.4 (appdir). Hopefully this can be addressed soon :D

**Comment by haneenmahd** [comment]
For my case, I think the font just loads fine but it's just the error when building the app. I'm using the `Newsreader` font. I'm also using the App Router, which is supposed to be stable by now.

**Comment by haneenmahd** [comment]
New Update!

This behaviour is random and happens unexpectedly. It doesn't seem like I have it right now! I didn't made any change to the code.


https://github.com/vercel/next.js/assets/72091386/de5ec62c-6587-49b1-8e11-b272ffef182d

**Comment by shaun-scale** [comment]
cc @leerob - lots of struggle here

**Comment by letkeeko** [comment]
+1 next font feature seems buggy 👎 - easier to go traditional?

**Comment by leerob** [comment]
Thanks for the shout @shaun-scale! We'll take a look 🙏

**Comment by daddotmn** [comment]
> > Another thing, I'm seeing that a lot of people including myself are having issues with Next new appdir in the different steps of developing an app. I would recommend that u don't use the app dir and use the common pages structure, I'm using v 13.4.1 that it's supposed to be stable and I'm breaking my head against the keyboad bc I can't get a font or a simple animation to work.
> 
> I've also been bashing my ahead against my keyboard, it's a real struggle after using the old system for so long 😅. Take this with a grain of salt as I've been very optimistic (and enthusiastic!) about this change for a while now, but:
> 
> `/app` is a massive shift in the Next paradigm and many other components in the ecosystem haven't quite adapted to the new format. In my opinion, the App directory is a lot faster and is just _so much more convenient_ to use! Personally, I make a lot of apps that pull data from APIs or databases, and the appDir just makes my life easier. Change usually takes a while, esp with smth as big as this, so it might take a while for third-party components and modules to become compatible with the new system.
> 
> In the mean time, you could wrap client-designed/legacy components in client component wrappers to have them work with the new system - I think the Next docs even talk about doing this (although I may be wrong).
> 
> Also, still having this issue with `Golos_Text` as of today on 13.4.4 (appdir). Hopefully this can be addressed soon :D

Same same

**Comment by jacobcwright** [comment]
+1 for `Instrument_Sans`

**Comment by haneenmahd** [comment]
> > > Another thing, I'm seeing that a lot of people including myself are having issues with Next new appdir in the different steps of developing an app. I would recommend that u don't use the app dir and use the common pages structure, I'm using v 13.4.1 that it's supposed to be stable and I'm breaking my head against the keyboad bc I can't get a font or a simple animation to work.
> > 
> > 
> > I've also been bashing my ahead against my keyboard, it's a real struggle after using the old system for so long 😅. Take this with a grain of salt as I've been very optimistic (and enthusiastic!) about this change for a while now, but:
> > `/app` is a massive shift in the Next paradigm and many other components in the ecosystem haven't quite adapted to the new format. In my opinion, the App directory is a lot faster and is just _so much more convenient_ to use! Personally, I make a lot of apps that pull data from APIs or databases, and the appDir just makes my life easier. Change usually takes a while, esp with smth as big as this, so it might take a while for third-party components and modules to become compatible with the new system.
> > In the mean time, you could wrap client-designed/legacy components in client component wrappers to have them work with the new system - I think the Next docs even talk about doing this (although I may be wrong).
> > Also, still having this issue with `Golos_Text` as of today on 13.4.4 (appdir). Hopefully this can be addressed soon :D
> 
> Same same

I have also noticed some weird bugs that are very hard to reproduce again but occurs often during development when using Next.js App Directory.

**Comment by haneenmahd** [comment]
> New Update!
> 
> This behaviour is random and happens unexpectedly. It doesn't seem like I have it right now! I didn't made any change to the code.
> 
> 
> https://github.com/vercel/next.js/assets/72091386/de5ec62c-6587-49b1-8e11-b272ffef182d

I can confirm that this issue still persists and is there a fix coming soon?

**Comment by haneenmahd** [comment]
> New Update!
> 
> This behaviour is random and happens unexpectedly. It doesn't seem like I have it right now! I didn't made any change to the code.
> 
> 
> https://github.com/vercel/next.js/assets/72091386/de5ec62c-6587-49b1-8e11-b272ffef182d

I can confirm this issue still persists and is there a new fix WIP?

I would be super happy to work on this issue if no one's currently working on one :)

**Comment by jacobcwright** [comment]
Any update or expected timeline?

**Comment by sozonome** [comment]
Same issue for `Geologica` font 🙏

**Comment by eihabkhan** [comment]
Any updates on this? Having the same issue with `Nanum Pen Script`

**Comment by evelynstender** [comment]
Any updates on this? I'm having the same issue with `Pathway Extreme` font.

**Comment by sofaemha** [comment]
> I don't know if it works on other devices or not. But in my personal project it works.

# TL;DR
Use a custom variable when declaring a NextFont component, then apply it as `className` in `<head>`. After that call custom variable in `global.css` file using `font-family`. For further documentation please refer to [Font Optimization](https://nextjs.org/docs/app/building-your-application/optimizing/fonts#using-multiple-fonts).

Example :
```tsx
// @/app/layout.tsx

import { Sono } from "next/font/google";

const sono = Sono({
  subsets: ["latin"],
  variable: '--custom-font-sono'
});

export default function Layout() {
  return (
    <html lang="en" className={ `${sono.variable}` }>
      <body>
        {children}
      </body>
    </html>
  );
}

```

```css
/* @/style/globals.css */

@tailwind base;
@tailwind components;
@tailwind utilities;

body {
  font-family: var(--custom-font-sono);
}

```

# ERROR CODE
## To Reproduce
> This is my code when there is an error.

```tsx
// @/app/layout.tsx

import Header from "@/components/Header";
import { Kumbh_Sans, Sono } from "next/font/google";

const kumbhsans = Kumbh_Sans({
  weight: "400",
  subsets: ["latin"]
});

const sono = Sono({
  subsets: ["latin"]
});

export default function RootLayout({ children }: { children: React.ReactNode }) {
  return (
    <html lang="en">
      <body className={ kumbhsans.className }>
        <Header className={ sono.className }/>
        {children}
      </body>
    </html>
  );
}

```

```tsx
// @/components/Header.tsx

export default function Header({ fontClass }: { fontClass: string }) {
  return (
    <header>
      <h1 className={ fontClass }>example</h1>
    </header>
  );
}

```

## Display Output
Output works as expected.

![{7E711288-F041-4134-9D33-915366583ED5} png](https://github.com/vercel/next.js/assets/75196975/e8774583-6958-41d6-9369-128c672d4e6e)

## Console Output
```bash
error - Failed to find font override values for font `Sono`
```

> I don't know why only one font is error.

# BRAINSTORMING

## Clue
Based on @BMscis comment on May 7th https://github.com/vercel/next.js/issues/47115#issuecomment-1537258403 .

## Idea
I was thinking what if the NextFont component is converted back in CSS and calls the font in CSS format.

## Solution
Then I found the answer in the [Font Optimization](https://nextjs.org/docs/app/building-your-application/optimizing/fonts#using-multiple-fonts) documentation.

# SOLVED CODE

## To Reproduce
> This is my code when I tried the idea.

```tsx
// @/app/layout.tsx

import Header from "@/components/Header";
import { Kumbh_Sans, Sono } from "next/font/google";

const kumbhsans = Kumbh_Sans({
  weight: "400",
  subsets: ["latin"],
  variable: '--custom-font-kumbhsans'
});

const sono = Sono({
  subsets: ["latin"],
  variable: '--custom-font-sono'
});

export default function RootLayout({ children }: { children: React.ReactNode }) {
  return (
    <html lang="en" className={ `${kumbhsans.variable} ${sono.variable}` }>
      <body>
        <Header />
        {children}
      </body>
    </html>
  );
}

```

```tsx
// @/components/Header.tsx

export default function Header() {
  return (
    <header>
      <h1>example</h1>
    </header>
  );
}

```

```css
/* @/style/globals.css */

@tailwind base;
@tailwind components;
@tailwind utilities;

body {
  font-family: var(--custom-font-kumbhsans);
}

h1 {
  font-family: var(--custom-font-sono);
}

```

## Display Output
Output works as expected.

![{7E711288-F041-4134-9D33-915366583ED5} png](https://github.com/vercel/next.js/assets/75196975/e8774583-6958-41d6-9369-128c672d4e6e)

**Comment by lukawski** [comment]
Hey, I started debugging it locally, here is what I found.

In my case I'm having issue with [Bodoni Moda](https://fonts.google.com/specimen/Bodoni+Moda).

Error I'm having is the same as other issue participants had:

```bash
- error Failed to find font override values for font `Bodoni Moda`
```
I added a new log to log the exact error that happens by modifying and adding below lines of code in `getFallbackFontOverrideMetrics` function inside `node_modules/next/dist/compiled/@next/font/dist/google/get-fallback-font-override-metrics.js`:
```typescript
    catch (err) {
        Log.error(`Failed to find font override values for font \`${fontFamily}\``);
        Log.error(err)
    }
```

Here is the error message
```bash
- error TypeError: Cannot destructure property 'category' of 'fontMetrics' as it is undefined.
    at calculateSizeAdjustValues (/node_modules/next/dist/server/font-utils.js:166:11)
    at getFallbackFontOverrideMetrics (/node_modules/next/dist/compiled/@next/font/dist/google/get-fallback-font-override-metrics.js:40:115)
```

So I went into `calculateSizeAdjustValues` function inside `node_modules/next/dist/server/font-utils.js` and added it rather clear that's the line causing the issue:
```typescript
    const fontMetrics = capsizeFontsMetrics[fontKey];
 ```
 
 For some reason this file `node_modules/next/dist/server/capsize-font-metrics.json` does not contain information about this particular font. So I went into it and checked - here is my font object, sitting in this json file:
 ```json
   "bodoniModa11pt": {
    "familyName": "Bodoni Moda 11pt",
    "category": "serif",
    "capHeight": 1500,
    "ascent": 2250,
    "descent": -800,
    "lineGap": 0,
    "unitsPerEm": 2000,
    "xHeight": 920,
    "xWidthAvg": 920
  },
```
Looks like it should be found? 

I decided to debug what `fonKey` is by adding 
```typscript
console.log({fontName, fontKey})
```
to previously mentioned `calculateSizeAdjustValues` function, here is the result:
```bash
{ fontName: 'Bodoni Moda', fontKey: 'bodoniModa' }
```

Looks like the issue is with `fontKey` value. I guess there are multiple possibilities why it doesn't work
- `fontKey` is not properly computed and doesn't include `11pt` 
- object held inside `capsize-font-metrics.json` file is not named correctly

In Google Font I see this font is named `Bodoni Moda` without 11pt suffix. Maybe it's just a leftover from some old implementation and changing it to `bodoniModa` is enough of a fix?

**Comment by LeandroBever** [comment]
`next/font` error:
Unknown font `Wix Madefor Display`

**Comment by VladimirCharkot** [comment]
Having this issue with 'Ysabeau', but only on the vercel deployment; I don't see the message in local development

**Comment by devBoysal** [comment]
I'm getting it on all the 'Ysabeau" fonts, no errors when you spin up the local server but once you refresh the page you can see in the terminal the following screenshot.

![image](https://github.com/vercel/next.js/assets/49185217/399da83c-efae-43eb-949d-5c20df7235f6)

**Comment by oseisaac** [comment]
Adding these 2 for the same font fixed the issue 

${instrumentFont.variable} ${instrumentFont.className}
```

```

**Comment by ChTayyab786** [comment]
same issue for "Instrument Sans" font.

<img width="535" alt="fonts-issue" src="https://github.com/vercel/next.js/assets/80338648/f9144959-30f8-4779-9458-6a11f4f85f40">

**Comment by nezzard** [comment]
Same Geologica

**Comment by xmarianox** [comment]
Same error with `Geologica`.

```shell
yarn run v1.22.19
$ next dev
  ▲ Next.js 13.5.6
  - Local:        http://localhost:3000

 ✓ Ready in 2.9s
 ⨯ Failed to find font override values for font `Geologica`
 ⨯ Failed to find font override values for font `Geologica`
 ✓ Compiled /page in 1610ms (471 modules)
 ✓ Compiled in 120ms (232 modules)
 ✓ Compiled in 60ms (232 modules)
 ✓ Compiled in 65ms (232 modules)
 ✓ Compiled in 55ms (232 modules)
```

**Comment by ngodacthien** [comment]
Try this everyone : "https://stackoverflow.com/questions/76478043/next-js-always-fail-at-downloading-fonts-from-google-fonts"

**Comment by olegshilov** [comment]
Same with `Handjet` font

**Comment by setamyDG** [comment]
same with `Tektur` font

**Comment by alstjd0051** [comment]
same issue with `Nanum Pen Script`

**Comment by mkaramuk** [comment]
Same with `Agbalumo` font

**Comment by ojpbarbosa** [comment]
Same with `Newsreader` font.

**Comment by joparedesm** [comment]
Kinda solved for me with **`display: 'swap'`** and **`adjustFontFallback: false`**  properties:

##  ⨯ Failed to find font override values for font \`Yuji Hentaigana Akari\` : Code

```js
import { Akaya_Telivigala, Yuji_Hentaigana_Akari, Lusitana } from 'next/font/google'

export const akaya = Akaya_Telivigala({
    weight:'400',
    subsets: ['latin']
});
export const yuji = Yuji_Hentaigana_Akari({
    weight:'400',
    subsets: ['latin']
});
export const lusitana = Lusitana({
    weight:['400', '700'],
    subsets: ['latin']
});
```
## Solution Code

```js
import { Akaya_Telivigala, Yuji_Hentaigana_Akari, Lusitana } from 'next/font/google'

export const akaya = Akaya_Telivigala({
    weight:'400',
    subsets: ['latin']
});
export const yuji = Yuji_Hentaigana_Akari({
    weight:'400',
    subsets: ['latin'],
    display: 'swap',
    adjustFontFallback: false
});
export const lusitana = Lusitana({
    weight:['400', '700'],
    subsets: ['latin']
});
```

Thought sometimes the font fails to load, well my theory is that the problem is within the size of some fonts that can't fit to Font fallback adjustment, this solution could work for small things but I can't recommend it to any real proyect, its better to ignore the fail massage if the font is loading allright

**Comment by shinyj1991** [comment]
same issue with `Noto Sans Thai Looped`

**Comment by mayognaise** [comment]
> Kinda solved for me with **`display: 'swap'`** and **`adjustFontFallback: false`** properties:

@joparedesm This saved my day!! Thank you so much!

**Comment by crnls** [comment]
Same with Onest font. Is there a permanent fix for this without resorting to display:swap?

**Comment by exelimpichment** [comment]
> @joparedesm

thank you mate

**Comment by nescheytt** [comment]
@joparedesm Thanx, it worked for me!

**Comment by DeMutombo** [comment]
@joparedesm Thank you this worked.

**Comment by samcx** [comment]
I can confirm this error message still shows for certain fonts (e.g., I tested Sonos)—possibly something is off with https://github.com/vercel/next.js/blob/335ff1cea9551bd847a398da4161142752d03a0c/packages/next/src/server/font-utils.ts#L95. It could also be that 

Using `adjustFontFallback: false` removes the error, which seems expected if we don't want to default to a fallback. Will <picture data-single-emoji=":dig-deep:" title=":dig-deep:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/dig-deep/de168db4e740f060.png" alt=":dig-deep:" align="absmiddle"></picture> into this soon!

**Comment by sspenst** [comment]
Same issue with Onest, resolved with the following:
```ts
export const onest = Onest({
  adjustFontFallback: false,
  display: 'swap',
  subsets: ['latin'],
});
```

**Comment by joaopcm** [comment]
Same for `Instrument Sans`

**Comment by muzucode** [comment]
Same for `Playfair`

**Comment by Super-Kenil** [comment]
+1 for `REM` fonts

**Comment by LiveWithCodeAnkit** [comment]
Same issue with Bricolage_Grotesque, resolved with the following:

export const secondaryFont = Bricolage_Grotesque({
  subsets: ["latin"],
  variable: "--font-grotesque",
  display: "swap",
  adjustFontFallback: false,
});

**Comment by ankitOpash** [comment]
> Same issue with Bricolage_Grotesque, resolved with the following:
> 
> ```
> export const secondaryFont = Bricolage_Grotesque({
>   subsets: ["latin"],
>   variable: "--font-grotesque",
>   display: "swap",
>   adjustFontFallback: false,
> });
> ```

**Comment by farhan-helmy** [comment]
Sames issue with `Bungee Spice`

**Comment by reissbaker** [comment]
FYI I'm pretty sure you don't need `display: 'swap'`; you just need `adjustFontFallback: false`. This happened to me for a font, and I have `display: 'block'`; simply setting `adjustFontFallback: false` resolved it, and I didn't need to set it to `swap`. (Personally, I aesthetically prefer the text to be invisible during the block period instead of being rendered in a different font; I find the visual shift, if there is one, to be less jarring when the layout goes from invisible-text-with-slightly-wrong-layout to correct instead of visible-text-in-the-wrong-font-*and*-the-wrong-layout to correct.)

It looks like the underlying issue is that [the default for adjustFontFallback is `true`](https://nextjs.org/docs/app/api-reference/components/font#adjustfontfallback), and that triggers the [getFallbackFontOverrideMetrics](https://github.com/vercel/next.js/blob/ad992d457681ed3ad0771fe2a23063a8801373dc/packages/font/src/google/get-fallback-font-override-metrics.ts#L13) function to be called, which eventually looks up the font in a precomputed JSON hash called capsize-font-metrics.json, which is pulled in a [task](https://github.com/vercel/next.js/blob/ad992d457681ed3ad0771fe2a23063a8801373dc/packages/next/taskfile.js#L116) from the [`@capsizecss/metrics/entireMetricsCollection`](https://github.com/seek-oss/capsize/blob/master/packages/metrics/src/entireMetricsCollection.json) package. If the Google font isn't in that precomputed metrics collection *at the version number that your version of NextJS is using*, it seems like the fallback font override will probably break.

**Comment by samcx** [comment]
Hi everyone! This fix is on the latest [canary](https://github.com/vercel/next.js/releases/tag/v14.1.1-canary.66) now (`v14.1.1-canary.66`).

Let us know if you are no longer seeing the font override error!

**Comment by masasidan** [comment]
> Hi everyone! This fix is on the latest [canary](https://github.com/vercel/next.js/releases/tag/v14.1.1-canary.66) now (`v14.1.1-canary.66`).
> 
> Let us know if you are no longer seeing the font override error!

Seems like it's now solved! Thanks👍

**Comment by Edward2362** [comment]
Sames issue with Grandiflora One

**Comment by Super-Kenil** [comment]
@Edward2362 did you check it in `v14.1.1-canary.66` ?

**Comment by ProchaLu** [comment]
@samcx, thx! We don't see any font override error with `Young_Serif` anymore. Fixed in `v14.1.1-canary.66`

**Comment by samcx** [comment]
Glad to hear everyone!

Will be closing this issue. We'll make sure to monitor the package version so this doesn't happen again (or push up a more automated solution)!

**Comment by ProchaLu** [comment]
> @samcx, thx! We don't see any font override error with Young_Serif anymore. Fixed in v14.1.1-canary.66

The issue got reintroduced in Next.js `v14.1.2` but fixed again in Next.js `v14.2.0-canary.0` 

```
   ▲ Next.js 14.1.2

   Creating an optimized production build ...
 ⨯ Failed to find font override values for font `Young Serif`
```

```
   ▲ Next.js 14.2.0-canary.0

   Creating an optimized production build ...
 ✓ Compiled successfully
```

**Comment by samcx** [comment]
@ProchaLu Are you able to reproduce reliably? May have been a stale `.next`.

A good reminder I should update the package again since a new version came out → https://github.com/seek-oss/capsize/releases.

**Comment by ProchaLu** [comment]
@samcx , here is a small [repro](https://github.com/ProchaLu/fonts-next) for this issue in Next.js `14.1.2`, but what I missed yesterday was that Next.js `14.1.2` [release notes](https://github.com/vercel/next.js/releases/tag/v14.1.2) say that this release does not include all pending features/changes on canary. 

> Note: this is a backport release for critical bug fixes -- this does not include all pending features/changes on canary

```
   ▲ Next.js 14.1.2
   - Experiments (use at your own risk):
     · typedRoutes

   Creating an optimized production build ...
 ⨯ Failed to find font override values for font `Young Serif`
 ```

**Comment by samcx** [comment]
@ProchaLu Ah I see. Yes, `14.1.2` does not have the <picture data-single-emoji=":pr:" title=":pr:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/pr/876af7e7d65e1595.png" alt=":pr:" align="absmiddle"></picture> with the fix. Just to clarify, it's perfectly fine to use the latest canary!

**Comment by webdiego** [comment]
Same...I followed the documentation..uff

**Comment by SiebeBaree** [comment]
![image](https://github.com/vercel/next.js/assets/33906209/635dc865-0799-4ac3-a8ac-9377ff400fe9)
Problem is still not fixed with the `Gabarito` google font. It doesn't give me an error but my font is still `system-ui`. I'm using `next@14.2.0-canary.47` (also tried 14.1.1 and 14.1.4)

![image](https://github.com/vercel/next.js/assets/33906209/b28974f8-ad6b-4475-baaf-cc8ebcc59e8d)

**Comment by webdiego** [comment]
In the end I have downloaded the font and am using it as I used to do...

**Comment by heysagnik** [comment]
> +1 for `Instrument_Sans`

any solutions ?

**Comment by samcx** [comment]
If you guys can provide <picture data-single-emoji=":repro:" title=":repro:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/repro/5cea53b5532825b4.png" alt=":repro:" align="absmiddle"></picture>s, I'll take a look!

**Comment by mwawrusch** [comment]
Still not fixed for `M PLUS Rounded 1c` in 14.2.0.

```
import { M_PLUS_Rounded_1c } from "next/font/google";
```

Thank you

**Comment by samcx** [comment]
@mwawrusch I think this particular one isn't supported. We use the package `@capsizecss/metrics` to find the font override.

- https://github.com/seek-oss/capsize/blob/ce7dcb7c75fa77f8dd84e17595edcc395fa0d657/packages/metrics/src/entireMetricsCollection.json#L14332
- Can confirm we're on the latest version → https://github.com/vercel/next.js/blob/c9c074c18d8d67f891672f361bc7084d512bec26/packages/next/package.json#L143

**Comment by mwawrusch** [comment]
> @mwawrusch I think this particular one isn't supported yet. We use the package `@capsizecss/metrics` to find the font override.
> 
> * https://github.com/seek-oss/capsize/blob/ce7dcb7c75fa77f8dd84e17595edcc395fa0d657/packages/metrics/src/entireMetricsCollection.json#L14332
> * Can confirm we're on the latest version → https://github.com/vercel/next.js/blob/c9c074c18d8d67f891672f361bc7084d512bec26/packages/next/package.json#L143

Thanks - I'll check on their site.

**Comment by mwawrusch** [comment]
@samcx 

Pull request should be merged soon: https://github.com/seek-oss/capsize/pull/192 . Appreciate all your work.

**Comment by samcx** [comment]
@mwawrusch The addition for `M PLUS Rounded 1c` should come up in the latest canary 🔜 !

**Comment by samcx** [comment]
It is on <picture data-single-emoji=":nextjs:" title=":nextjs:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/nextjs/2a85d633ae594556.png" alt=":nextjs:" align="absmiddle"></picture> [v14.3.0-canary.4](https://github.com/vercel/next.js/releases/tag/v14.3.0-canary.4) now!

**Comment by pablojsx** [comment]
Same problem

**Comment by samcx** [comment]
Looks like this locked due to inactivity. I will be constantly looking to update the `@capsizecss/metrics` package until we automate it!

@kxhcfk We may have to trigger another one of this → https://github.com/vercel/next.js/pull/62704, looking into it.
